### PR TITLE
glibc: additionally provide -lcrypt

### PR DESCRIPTION
**Pull request by andrewrk**
Looks like we were missing support for this other library that glibc provides.

### CODE REVIEW

File: tools/update_glibc.zig [hunk 1]
```diff
@@ -22,6 +22,7 @@ const lib_names = [_][]const u8{
     "rt",
     "ld",
     "util",
+    "crypt",
```
Comment (daurnimator) [inline tools/update_glibc.zig:25]: alphabetical order?

**Review by daurnimator — COMMENTED** [review]
There might be other missing glibc libraries too?
```
$ pacman -Qql glibc | grep so | grep 2.32
/usr/lib/ld-2.32.so
/usr/lib/libBrokenLocale-2.32.so
/usr/lib/libanl-2.32.so
/usr/lib/libc-2.32.so
/usr/lib/libcrypt-2.32.so
/usr/lib/libdl-2.32.so
/usr/lib/libm-2.32.so
/usr/lib/libmvec-2.32.so
/usr/lib/libnsl-2.32.so
/usr/lib/libnss_compat-2.32.so
/usr/lib/libnss_db-2.32.so
/usr/lib/libnss_dns-2.32.so
/usr/lib/libnss_files-2.32.so
/usr/lib/libnss_hesiod-2.32.so
/usr/lib/libpthread-2.32.so
/usr/lib/libresolv-2.32.so
/usr/lib/librt-2.32.so
/usr/lib/libutil-2.32.so
```

libnss and ld you can probably skip; but the others....

**Comment by daurnimator** [comment]
FYI: 
> new glibc drops libcrypt.h header and corresponding static library. The upstream plans to eventually remove libcrypt from glibc tree in favor of a separate project libxcrypt
> There's no breakage (famous last words) as glibc still ships libcrypt.so.1 for backwards compatibility, while libxcrypt package provides development headers and library with bumped soname version due to removal of the functions providing insecure hashes. You may need to add '#include <crypt.h>' if it was missing before as uninstd.h no longer provides a declaration of the crypt function.

**Comment by andrewrk** [comment]
Ahh that's good to know. Thanks. In this case I think it should be reverted for simplicity's sake.

**Comment by daurnimator** [comment]
> In this case I think it should be reverted for simplicity's sake.

maybe not?
As long as glibc keeps providing libcrypt.so.1 for backwards compatibility, we should too, right?

**Comment by andrewrk** [comment]
Hmm, fair point. :thinking:

**Comment by Jan200101** [comment]
on Fedora libcrypt.so.1 has already been dropped from the glibc package and is instead provided by libxcrypt

**Comment by jedisct1** [comment]
Same on Arch Linux.

**Comment by andrewrk** [comment]
6b7ddfbafe7ff8f9bc9952f838737e8514c9b26b

**Pull request merged**